src/Servers/DoHServer.php

```diff
@@ -17,6 +25,20 @@ public function call( $dnsQuery ): ?string {
 				curl_setopt( $ch, CURLOPT_CONNECTTIMEOUT, 10 );
 				curl_setopt( $ch, CURLOPT_TIMEOUT, 10 );
 				curl_setopt( $ch, CURLOPT_HTTPHEADER, $headers );
+				if (count($this->ips) != 0) {
+					$parsed = parse_url($this->url, PHP_URL_HOST);
+					if ($parsed === false) {
+						throw new InvalidArgumentException( 'URL cannot parse' );
```

Comment:
I wasn't sure how much should I worry about potentially invalid URL here; but; nothing user-generated should be put here directly without validation anyway. (That should maybe go to the readme btw.)